types/braintree/index.d.ts

```diff
@@ -0,0 +1,1638 @@
+    }
+
+    export interface MerchantFunding {
+        accountNumber?: string;
+        accountNumberLast4?: string;
+        descriptor?: string;
+        destination: string;
+        email?: string;
+        mobilePhone?: string;
+        routingNumber?: string;
+    }
+
+    export interface MerchantIndividual {
+        addressDetails: MerchantAddressDetails;
+        dateOfBirth: string;
+        email: string;
+        firstName: string;
+        lastName: string;
+        phone?: string;
+        ssn?: string;
+        ssnLast4?: string;
+    }
+
+    export type MerchantAccountStatus = 'Pending' | 'Active' | 'Suspended';
+
+    /**
+     * Payment Method
+     */
+
+    // Payment method is an instance of one of these types
+    export type PaymentMethod = AndroidPayCard | ApplePayCard | PayPalAccount | CreditCard | SamsungPayCard |
+        VenmoAccount | VisaCheckoutCard | MasterpassCard;
+
+    interface PaymentMethodRequest {
+        billingAddressId?: string;
+        cardholderName?: string;
+        cvv?: string;
+        deviceData?: string;
+        expirationDate?: string;
+        expirationMonth?: string;
+        expirationYear?: string;
+        number?: string;
+        paymentMethodNonce: string;
+    }
+
+    export interface PaymentMethodCreateRequest extends PaymentMethodRequest {
+        billingAddress?: {
+            company?: string;
+            countryCodeAlpha2?: string;
+            countryCodeAlpha3?: string;
+            countryCodeNumeric?: string;
+            countryName?: string;
+            extendedAddress?: string;
+            firstName?: string;
+            lastName?: string;
+            locality?: string;
+            postalCode?: string;
+            region?: string;
+            streetAddress?: string;
+        };
+        customerId: string;
+        options?: {
+            failOnDuplicatePaymentMethod?: boolean;
+            makeDefault?: boolean;
+            verificationAcmount?: string;
+            verificationMerchantAccountId?: string;
+            verifyCard?: boolean;
+        };
+    }
+
+    export interface PaymentMethodUpdateRequest extends PaymentMethodRequest {
+        billingAddress?: {
+            company?: string;
+            countryCodeAlpha2?: string;
+            countryCodeAlpha3?: string;
+            countryCodeNumeric?: string;
+            countryName?: string;
+            extendedAddress?: string;
+            firstName?: string;
+            lastName?: string;
+            locality?: string;
+            postalCode?: string;
+            region?: string;
+            streetAddress?: string;
+            options?: {
+                updateExisting?: boolean
+            }
+        };
+        options?: {
+            makeDefault?: boolean;
+            verificationAcmount?: string;
+            verificationMerchantAccountId?: string;
+            verifyCard?: boolean;
+        };
+    }
+
+    /**
+     * Payment Method Nonce
+     */
+
```

Comment:
Which Braintree docs are you looking at for this?  I just picked this part at random, but it doesn't seem to match up with what I see here:

https://developers.braintreepayments.com/reference/response/payment-method-nonce/node

Reply:
@jasonvideoblocks I think it looks the same. I setup the nested fields as interfaces so that might be why you see fewer top-level props in the ts definition. Let me know if a specific field I missed though.

Reply:
🤦‍♂ 

I was misreading Braintree's docs, I didn't realize those were nested attributes.  So, uh, never mind and carry on.  😆